examples/beta/openai/realtime/audio_stream_input_output_tools.py

```diff
@@ -0,0 +1,44 @@
+import asyncio
+from io import BytesIO
+
+from typing import AsyncGenerator, Any
+
+from pydub import AudioSegment
+from pydub.playback import play
+
+from mirascope.beta.openai import Realtime, record_as_stream, OpenAIRealtimeTool
+
+app = Realtime(
+    "gpt-4o-realtime-preview-2024-10-01",
```

Comment:
the user should also be able to set tools at the top level here that will be set in the initial `session.update`

Reply:
It is already supported.
```
class Realtime:
    def __init__(
        self,
        model: str,
        *,
        context: dict[str, Any] | None = None,
        modalities: list[str] | NotSet = NotSet,
        instructions: str | NotSet = NotSet,
        voice: str | NotSet = NotSet,
        turn_detection: dict[str, Any] | None | NotSet = NotSet,
        input_audio_format: str | NotSet = NotSet,
        output_audio_format: str | NotSet = NotSet,
        input_audio_transcription: str | NotSet = NotSet,
        tool_choice: str | NotSet = NotSet,
        temperature: float | NotSet = NotSet,
        max_response_output_tokens: int | float | Literal["inf"] = NotSet,
        tools: list[type[OpenAIRealtimeTool] | Callable] | NotSet = NotSet,
    ) -> None:
 ```

Reply:
it's supported